client/command/socks.go

```diff
@@ -0,0 +1,130 @@
+package command
+
+import (
+	"context"
+	"fmt"
+	"github.com/bishopfox/sliver/protobuf/sliverpb"
+	"io"
+	"log"
+	"net"
+
+	"github.com/bishopfox/sliver/client/core"
+	"github.com/bishopfox/sliver/protobuf/rpcpb"
+
+	"github.com/desertbit/grumble"
+)
+
+func socks(ctx *grumble.Context, rpc rpcpb.SliverRPCClient) {
+	session := ActiveSession.GetInteractive()
+	if session == nil {
+		return
+	}
+
+	port := ctx.Flags.Int("port")
+	go startSocksServer(ctx, port, rpc)
+}
+
+func startSocksServer(ctx *grumble.Context, port int, rpc rpcpb.SliverRPCClient) {
+	fmt.Printf(Info + fmt.Sprintf("Starting socks server on port %d\n\n", port))
+	session := ActiveSession.Get()
+	if session == nil {
+		fmt.Println("Current session is none")
+		return
+	}
+
+	netAddr, err := net.ResolveTCPAddr("tcp4", fmt.Sprintf("0.0.0.0:%d", port))
```

Comment:
I know this is a WIP, but just a heads up: we probably want to make this IP address configurable by the user, so that they can listen on a specific interface. Defaulting to `0.0.0.0` is a good option though.

Reply:
Sure, it is far from PR ready

Reply:
Thanks 😄

Reply:
My previous comment still stands :)